src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -192,9 +192,8 @@ public async Task<int> DeployAsync()
                         throw new ValidationException($"Could not retrieve the Batch account name from virtual machine {configuration.VmName}.");
                     }
 
-                    batchAccount = (await new BatchManagementClient(tokenCredentials) { SubscriptionId = configuration.SubscriptionId }.BatchAccount.ListByResourceGroupAsync(configuration.ResourceGroupName))
-                        .FirstOrDefault(a => a.Name.Equals(batchAccountName, StringComparison.OrdinalIgnoreCase))
-                            ?? throw new ValidationException($"Batch account {batchAccountName} does not exist in resource group {configuration.ResourceGroupName}.");
+                    batchAccount = await TryGetExistingBatchAccountAsync(batchAccountName)
```

Comment:
These two TryGet methods don't use try-get pattern, so I'd suggest dropping "Try" in the name.

Reply:
Agreed - the usual try-get pattern is not allowed in async functions.

Reply:
changed to GetExistingStorageAccountAsync